.github/workflows/build-laghos.yml

```diff
@@ -0,0 +1,106 @@
+  HYPRE_TOP_DIR: hypre-2.11.2
+  METIS_URL: http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/OLD
+  METIS_ARCHIVE: metis-4.0.3.tar.gz
+  METIS_TOP_DIR: metis-4.0.3
+  MFEM_REPO: https://github.com/mfem/mfem.git
+  MFEM_BRANCH: master
+  MFEM_TOP_DIR: mfem
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    # Checkout Laghos in "laghos" subdirectory. Final path: /home/runner/work/ceed/laghos/laghos
+    # Note: it must be done now to access "install-hypre", "install-metis" and "install-mfem" actions.
+    - name: checkout Laghos
+      uses: actions/checkout@v2
+      with:
+        path: laghos
+
+    - name: Get MPI (Linux)
+      run: |
+        sudo apt-get install mpich libmpich-dev
+        export MAKE_CXX_FLAG="MPICXX=mpic++"
+
+    # Get Hypre through cache, or install it.
+    # Install will only run on cache miss.
+    - name: Cache Hypre Install
+      id: hypre-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.HYPRE_TOP_DIR }}
+        key: ${{ runner.os }}-build-${{ env.HYPRE_TOP_DIR }}-v1
+
+    - name: Get Hypre
+      if: steps.hypre-cache.outputs.cache-hit != 'true'
+      uses: ./laghos/.github/actions/install-hypre
+      with:
+        hypre-archive: ${{ env.HYPRE_ARCHIVE }}
+        hypre-dir: ${{ env.HYPRE_TOP_DIR }}
+
+    # Get Metis through cache, or install it.
+    # Install will only run on cache miss.
+    - name: Cache Metis Install
+      id: metis-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.METIS_TOP_DIR }}
+        key: ${{ runner.os }}-build-${{ env.METIS_TOP_DIR }}-v1
+
+    - name: Install Metis
+      if: steps.metis-cache.outputs.cache-hit != 'true'
+      uses: ./laghos/.github/actions/install-metis
+      with:
+        metis-url: ${{ env.METIS_URL }}
+        metis-archive: ${{ env.METIS_ARCHIVE }}
+        metis-dir: ${{ env.METIS_TOP_DIR }}
+
+    # make generic links to libraries for MFEM install
+    - name: configure links
+      run: |
+        echo "Hypre symlink:"
+        ln -s $HYPRE_TOP_DIR hypre;
+        echo "Metis symlink:"
+        ln -s $METIS_TOP_DIR metis-4.0;
+
+    - name: MFEM master commit
+      run: |
+        echo "MFEM_COMMIT=$(git ls-remote --heads ${{ env.MFEM_REPO }} ${{ env.MFEM_BRANCH }} | awk '{print $1;}')" >> $GITHUB_ENV
+
+    # Get MFEM through cache, or install it.
+    # Install will only run on cache miss.
+    - name: Cache MFEM Install
+      id: mfem-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.MFEM_TOP_DIR }}
+        key: ${{ runner.os }}-build-${{ env.MFEM_TOP_DIR }}-${{ env.MFEM_COMMIT }}-v2
+
+    - name: Install MFEM
+      if: steps.mfem-cache.outputs.cache-hit != 'true'
+      uses: ./laghos/.github/actions/install-mfem
+      with:
+        mfem-repo: ${{ env.MFEM_REPO }}
+        mfem-branch: ${{ env.MFEM_BRANCH }}
+        mfem-dir: ${{ env.MFEM_TOP_DIR }}
+
+    - name: make
+      run: |
+        cd laghos && make -j
+    - name: checks 1
```

Comment:
Would it be possible to split the build/checks/tests in different actions?
If not , could we rename this one from `build_laghos` to `tests_laghos` or just `tests`?

Reply:
I'm trying to understand exactly what it is you would like.
Do you mean that in the checks view you would like to have separate logs for each?

In terms of wording, 
- an "action" is just an encapsulated... action, that can be re-used elsewhere.
- in the check view, _build_ is a "job" with multiples "steps" in it, it should in fact be _build_and_test_
- _build_laghos_ is the "workflow" name. It should certainly be renamed _build_and_test_laghos_

In the end, I could try putting checks and tests in a different job. However I think that would require passing the installs as artifact, which may add complexity.

I can give it a try, let me know if that's what you meant.

Reply:
`build_and_test_laghos` would be perfect.